Source/WebCore/css/parser/CSSSupportsParser.cpp

```diff
@@ -147,13 +147,32 @@ CSSSupportsParser::SupportsResult CSSSupportsParser::consumeSupportsSelectorFunc
     return CSSSelectorParser::supportsComplexSelector(block, m_parser.context(), m_isNestedContext) ? Supported : Unsupported;
 }
 
+
+CSSSupportsParser::SupportsResult CSSSupportsParser::consumeSupportsFontFormatFunction(CSSParserTokenRange& range)
+{
+    if (range.peek().type() != FunctionToken || range.peek().functionId() != CSSValueFontFormat)
+        return Invalid;
+
+    auto block = range.consumeBlock();
+    block.consumeWhitespace();
+
+    auto isSupported = CSSPropertyParserHelpers::consumeIdent(block, CSSPropertyParserHelpers::identMatchesSupportedFontFormat) ? Supported : Unsupported;
+    if (!block.atEnd())
+        return Unsupported;
+
+    return isSupported;
+}
+
 CSSSupportsParser::SupportsResult CSSSupportsParser::consumeConditionInParenthesis(CSSParserTokenRange& range,  CSSParserTokenType startTokenType)
 {
     // <supports-in-parens> = ( <supports-condition> ) | <supports-feature> | <general-enclosed>
     if (startTokenType == IdentToken && range.peek().type() != LeftParenthesisToken) {
         if (range.peek().type() == FunctionToken && range.peek().functionId() == CSSValueSelector)
             return consumeSupportsSelectorFunction(range);
 
+        if (range.peek().type() == FunctionToken && range.peek().functionId() == CSSValueFontFormat)
```

Comment:
It doesn't seem like this check is strictly needed, since you check this and return `Invalid` if it doesn't meet these requirements. Seems like you're doing the same work twice.

It seems like this would be entirely equivalent:
```
    return consumeSupportsFontFormatFunction(range);
```

Alternatively, if you think you will need to add more range checks in the future, maybe we should change the internal check to an ASSERT, since we expect the check to happen before entering the method.